Formula/postgresql@12.rb

```diff
@@ -81,22 +81,100 @@ def post_install
     return if ENV["CI"]
 
     (var/"log").mkpath
-    (var/"postgres").mkpath
-    unless File.exist? "#{var}/postgres/PG_VERSION"
-      system "#{bin}/initdb", "--locale=C", "-E", "UTF-8", "#{var}/postgres"
+    versioned_postgresql_data_dir.mkpath
+    unless versioned_data_dir_exists?
+      system "#{bin}/initdb", "--locale=C", "-E", "UTF-8", versioned_postgresql_data_dir
     end
   end
 
+  # Previous versions of this formula used the same data dir as the regular
+  # postgresql formula. So we check whether the versioned data dir exists
+  # and has a PG_VERSION file, which should indicate that the versioned
+  # data dir is in use. Otherwise, returns the old data dir path.
+  def postgresql_datadir
+    if versioned_data_dir_exists?
+      versioned_postgresql_data_dir
+    else
+      old_postgresql_data_dir
+    end
+  end
+
+  def versioned_postgresql_data_dir
+    "#{var}/#{name}"
+  end
+
+  def old_postgresql_data_dir
+    "#{var}/postgres"
+  end
+
+  # Same as with the data dir - use old log file if the old data dir
+  # is version 12
+  def postgresql_log_path
+    if versioned_data_dir_exists?
+      "#{var}/log/#{name}"
+    else
+      "#{var}/log/postgres"
+    end
+  end
+
+  def versioned_data_dir_exists?
+    File.exist?("#{versioned_postgresql_data_dir}/PG_VERSION")
```

Comment:
```suggestion
   ( versioned_postgresql_data_dir/"PG_VERSION").exist?
```

Reply:
Addressed in a608c8ba518efb1e88325c731817788bd29507b9